src/app/v2/inventory/projects/[projectId]/actions.ts

```diff
@@ -0,0 +1,37 @@
+"use server";
+import { createProjectListClient } from "@/lib/rpc/client";
+import { sessionMustGetTenant } from "@/lib/session/session";
+import { getAccessToken } from "@auth0/nextjs-auth0";
+
+export async function listProjectVersions(projectId: string) {
+  const { accessToken } = await getAccessToken();
+  if (!accessToken) {
+    throw new Error("No access token found");
+  }
+  const tenant = await sessionMustGetTenant();
+  const service = createProjectListClient(tenant, accessToken);
+  return await service.listProjectVersions({ projectId });
+}
+
+export async function listProjectVersionBOM(versionId: string) {
+  const { accessToken } = await getAccessToken();
```

Comment:
We are duplicating the code for getting the access token, tenant and creating the project service before calling RPC within the project service. We should have a single function that creates the project service that we can use in different operations such as `listProjectVersions`, `listProjectVersionBOM`, `listBOMComponents`